src/main/java/net/dv8tion/jda/internal/entities/GuildImpl.java

```diff
@@ -234,6 +242,76 @@ public void invalidate()
         }
     }
 
+    public void uncacheChannel(GuildChannel channel)
+    {
+        long id = channel.getIdLong();
+        switch (channel.getType())
+        {
+        case TEXT:
+            api.getTextChannelsView().remove(id);
+            this.getTextChannelsView().remove(id);
+            break;
+        case NEWS:
+            api.getNewsChannelView().remove(id);
+            this.getNewsChannelView().remove(id);
+            break;
+        case MEDIA:
+            api.getMediaChannelsView().remove(id);
+            this.getMediaChannelsView().remove(id);
+            break;
+        case FORUM:
+            api.getForumChannelsView().remove(id);
+            this.getForumChannelsView().remove(id);
+            break;
+        case VOICE:
+            api.getVoiceChannelsView().remove(id);
+            this.getVoiceChannelsView().remove(id);
+            break;
+        case STAGE:
+            api.getStageChannelView().remove(id);
+            this.getStageChannelsView().remove(id);
+            break;
+        case CATEGORY:
+            api.getCategoriesView().remove(id);
+            this.getCategoriesView().remove(id);
+            break;
+        case GUILD_NEWS_THREAD:
+        case GUILD_PUBLIC_THREAD:
+        case GUILD_PRIVATE_THREAD:
+            api.getThreadChannelsView().remove(id);
+            this.getThreadChannelsView().remove(id);
+            break;
+        }
+
+        // Remove dangling threads
+        if (channel instanceof IThreadContainer)
+        {
+            SortedSnowflakeCacheViewImpl<ThreadChannel> localView = this.getThreadChannelsView();
+            SnowflakeCacheViewImpl<ThreadChannel> globalView = api.getThreadChannelsView();
+            Predicate<ThreadChannel> predicate = thread -> channel.equals(thread.getParentChannel());
```

Comment:
Should this match on parentChannelId instead of parentChannel since `getParentChannel` looks at the cache and the parent has already been removed from the cache?

Reply:
That doesn't matter since the channel is locally cached in the thread, and only updated if the cache has a newer non-null instance available.